src/ndb_common.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Roman Gezikov <rgezikov@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef NDB_COMMON_H_
+#define NDB_COMMON_H_
+
+#include <libswiftnav/common.h>
+#include <libswiftnav/almanac.h>
+#include <libswiftnav/ephemeris.h>
+
+typedef u32 ndb_file_version_t;
+
+enum ndb_op_code
+{
+    NDB_ERR_NONE = 0,              /**< No error */
+    NDB_ERR_UNSUPPORTED_IE,        /**< Current platform doesn't support this IE */
+    NDB_ERR_MISSING_IE,            /**< DB doesn't contain value of this IE */
+    NDB_ERR_UNSUPPORTED_FUNCTION,
+    NDB_ERR_FILE_IO,
+    NDB_ERR_OP_FAILURE,
+    NDB_ERR_NO_SECTION_SPACE,
+    NDB_ERR_SECTION_INIT_FAILURE,
+    NDB_ERR_INIT_DONE
+};
+
+enum ndb_data_source
+{
+    NDB_DS_UNDEFINED, NDB_DS_RECEIVER, NDB_DS_SBP
+};
+
+enum ndb_information_element
```

Comment:
Ok, I am reasonably confident my second paragraph above is accurate: The benefit of the more complicated approach is that all the NDB functions exist on all hardware platforms, but unsupported data would yield `NDB_ERR_UNSUPPORTED_FUNCTION`. I do think this will be useful in a handful of places (e.g. L2C capability), but there are easier and more performant ways of getting the same functionality. Worst case, we could manually implement or #define unsupported functions to return an error. Better, we could add something like `__attribute__ ((weak, alias("ndb_unimplemented")))` to all the public NDB API functions. That way anything that isn't resolved by the linker will be aliased to a function which returns `NDB_ERR_UNSUPPORTED_FUNCTION`.